### DISCUSSION TITLE
Error when joining stiffnessmatrix and mesh

### DISCUSSION THREAD

**Discussion by recepkaya7** [kinnala/scikit-fem#1019] [Q&A]
<meshio mesh object>
 Number of points: 339590
 Number of cells:
 tetra: 1263094
 
**I have a mesh file with .vtk extension with properties. When combining the stiffness matrix with my mesh file, I get this error:**
  ---------------------------------------------------------------- -------------------------
AttributeError Traceback (most recent call last)
Input In [20], in <cell line: 33>()
```
      30 return density * u * v * w
      32# Assemble matrices
---> 33 A = stiffness.assemble(mesh)
      34 M = mass.assemble(mesh)
      38 # Define the load

```

File ~\anaconda3\lib\site-packages\skfem\assembly\form\biliear_form.py:146, in
```
 BilinearForm.assemble(self, *args, **kwargs)
     144 assert self.form is not None
     145 logger.info("Assembling '{}'.".format(self.form.__name__))
--> 146 mat = COOOData._assemble_scipy_csr(*self._assemble(*args, **kwargs))
     147 logger.info("Assembling finished.")
     148 return mat
```

File ~\anaconda3\lib\site-packages\skfem\assembly\form\biliear_form.py:72, in 

```BilinearForm._assemble(self, ubasis, vbasis, **kwargs)
      68 elif ubasis.X.shape[-1] != vbasis.X.shape[-1]:
      69 raise ValueError("Quadrature mismatch: trial and test functions "
      70 "should have the same number of integration points.")
---> 72 nt = ubasis.nelems
      73 dx = ubasis.dx
      74 wdict = FormExtraParams({
      75 **ubasis.default_parameters(),
      76 **self._normalize_asm_kwargs(kwargs, ubasis),
      77 })
```

**AttributeError: 'Mesh' object has no attribute 'nelems'.**

**How can I solve this problem?
I'm new to this and would appreciate your help..**

**Comment by kinnala** [comment]
Which mesh type you have? Triangular, tetrahedral, quadrilateral? Can you try to replace

A = stiffness.assemble(mesh)

by

A = stiffness.assemble(Basis(mesh, ElementTriP1()))

You need to pick correct element for your mesh. One of the following:

ElementTriP1
ElementTetP1
ElementQuad1
ElementHex1
